filebeat/input/filestream/fswatch.go

```diff
@@ -70,27 +56,22 @@ type fileWatcherConfig struct {
 // fileWatcher gets the list of files from a FSWatcher and creates events by
 // comparing the files between its last two runs.
 type fileWatcher struct {
-	interval        time.Duration
-	resendOnModTime bool
-	prev            map[string]os.FileInfo
-	scanner         loginp.FSScanner
-	log             *logp.Logger
-	events          chan loginp.FSEvent
-	sameFileFunc    func(os.FileInfo, os.FileInfo) bool
+	cfg     fileWatcherConfig
+	prev    map[string]loginp.FileDescriptor
+	scanner loginp.FSScanner
+	log     *logp.Logger
+	events  chan loginp.FSEvent
 }
 
 func newFileWatcher(paths []string, ns *conf.Namespace) (loginp.FSWatcher, error) {
+	var config *conf.C
 	if ns == nil {
-		return newScannerWatcher(paths, conf.NewConfig())
+		config = conf.NewConfig()
+	} else {
+		config = ns.Config()
 	}
 
-	watcherType := ns.Name()
-	f, ok := watcherFactories[watcherType]
-	if !ok {
-		return nil, fmt.Errorf("no such file watcher: %s", watcherType)
-	}
-
-	return f(paths, ns.Config())
```

Comment:
Just wondering why the factory code needed to be removed as part of this PR.  Not too familiar with this codebase so I may be missing something obvious.

Reply:
I removed because it was a useless abstraction that only made it confusing. There is neither other implementation of the prospector nor scanner, the factory was used for mocking the scanner in unit tests. Since I've rewritten all the tests from scratch using real files, it's not needed anymore, hence removed.

Reply:
Got it, makes sense.  Thanks for the explanation.